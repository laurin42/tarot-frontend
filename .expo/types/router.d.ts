/* eslint-disable */
import * as Router from 'expo-router';

export * from 'expo-router';

declare module 'expo-router' {
  export namespace ExpoRouter {
    export interface __routes<T extends string | object = string> {
      hrefInputParams: { pathname: Router.RelativePathString, params?: Router.UnknownInputParams } | { pathname: Router.ExternalPathString, params?: Router.UnknownInputParams } | { pathname: `/app`; params?: Router.UnknownInputParams; } | { pathname: `/_sitemap`; params?: Router.UnknownInputParams; } | { pathname: `${'/(tabs)'}/dailyCard` | `/dailyCard`; params?: Router.UnknownInputParams; } | { pathname: `${'/(tabs)'}` | `/`; params?: Router.UnknownInputParams; } | { pathname: `${'/(tabs)'}/profile` | `/profile`; params?: Router.UnknownInputParams; } | { pathname: `${'/(tabs)'}/threeCards` | `/threeCards`; params?: Router.UnknownInputParams; } | { pathname: `/+not-found`, params: Router.UnknownInputParams & {  } };
      hrefOutputParams: { pathname: Router.RelativePathString, params?: Router.UnknownOutputParams } | { pathname: Router.ExternalPathString, params?: Router.UnknownOutputParams } | { pathname: `/app`; params?: Router.UnknownOutputParams; } | { pathname: `/_sitemap`; params?: Router.UnknownOutputParams; } | { pathname: `${'/(tabs)'}/dailyCard` | `/dailyCard`; params?: Router.UnknownOutputParams; } | { pathname: `${'/(tabs)'}` | `/`; params?: Router.UnknownOutputParams; } | { pathname: `${'/(tabs)'}/profile` | `/profile`; params?: Router.UnknownOutputParams; } | { pathname: `${'/(tabs)'}/threeCards` | `/threeCards`; params?: Router.UnknownOutputParams; } | { pathname: `/+not-found`, params: Router.UnknownOutputParams & {  } };
      href: Router.RelativePathString | Router.ExternalPathString | `/app${`?${string}` | `#${string}` | ''}` | `/_sitemap${`?${string}` | `#${string}` | ''}` | `${'/(tabs)'}/dailyCard${`?${string}` | `#${string}` | ''}` | `/dailyCard${`?${string}` | `#${string}` | ''}` | `${'/(tabs)'}${`?${string}` | `#${string}` | ''}` | `/${`?${string}` | `#${string}` | ''}` | `${'/(tabs)'}/profile${`?${string}` | `#${string}` | ''}` | `/profile${`?${string}` | `#${string}` | ''}` | `${'/(tabs)'}/threeCards${`?${string}` | `#${string}` | ''}` | `/threeCards${`?${string}` | `#${string}` | ''}` | { pathname: Router.RelativePathString, params?: Router.UnknownInputParams } | { pathname: Router.ExternalPathString, params?: Router.UnknownInputParams } | { pathname: `/app`; params?: Router.UnknownInputParams; } | { pathname: `/_sitemap`; params?: Router.UnknownInputParams; } | { pathname: `${'/(tabs)'}/dailyCard` | `/dailyCard`; params?: Router.UnknownInputParams; } | { pathname: `${'/(tabs)'}` | `/`; params?: Router.UnknownInputParams; } | { pathname: `${'/(tabs)'}/profile` | `/profile`; params?: Router.UnknownInputParams; } | { pathname: `${'/(tabs)'}/threeCards` | `/threeCards`; params?: Router.UnknownInputParams; } | `/+not-found` | { pathname: `/+not-found`, params: Router.UnknownInputParams & {  } };
    }
  }
}
