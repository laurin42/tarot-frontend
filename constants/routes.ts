import type { Route } from '@/types/navigation';

/**
 * Typsichere Routenkonstanten für die Navigation
 */
export const APP_ROUTES = {
  TABS: {
    THREECARDS: "/(tabs)/threecards" as Route,
    DAILYCARD: "/(tabs)/dailycard" as Route,
    PROFILE: "/(tabs)/profile" as Route,
  },
  AUTH: "/(auth)" as Route,
  HOME: "/" as Route,
};

/**
 * Konstanten für dynamische Routen
 */
export const DYNAMIC_ROUTES = {
  DEV_TOOLS: "/dev-tools" as const,
};