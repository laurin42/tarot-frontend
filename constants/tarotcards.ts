export interface ITarotCard {
  id: string;
  name: string;
  image: any;
}

export interface ISelectedAndShownCard extends ITarotCard {
  name: string;
  showFront?: boolean;
  isSelected?: boolean;
  currentIndex?: number;
  explanation?: string;
  image: any;
  onNextCard: () => void;
}

export const tarotCards: ITarotCard[] = [
  {
    id: "0",
    name:  "Der Narr",
    image: require("@/assets/images/tarot_cards/The_fool.jpg"),
  },
  {
    id: "1",
    name:  "Der Magier",
    image: require("@/assets/images/tarot_cards/The_magician.jpg"),
  },
  {
    id: "2",
    name: "Die Kaiserin",
    image: require("@/assets/images/tarot_cards/The_empress.jpg"),
  },
  {
    id: "3",
    name: "Die Hohepriesterin",
    image: require("@/assets/images/tarot_cards/High_priestess.jpg"),
  },
  {
    id: "4",
    name: "König der Kelche",
    image: require("@/assets/images/tarot_cards/King_of_cups.jpg"),
  },
  {
    id: "5",
    name: "Ritter der Kelche",
    image: require("@/assets/images/tarot_cards/Knight_of_cups.jpg"),
  },
  {
    id: "6",
    name: "Ritter der Münzen",
    image: require("@/assets/images/tarot_cards/Knight_of_pentacles.jpg"),
  },
  {
    id: "7",
    name: "Page der Kelche",
    image: require("@/assets/images/tarot_cards/Page_of_cups.jpg"),
  },
  {
    id: "8",
    name: "Page der Münzen",
    image: require("@/assets/images/tarot_cards/Page_of_pentacles.jpg"),
  },
  {
    id: "9",
    name: "Königin der Kelche",
    image: require("@/assets/images/tarot_cards/Queen_of_cups.jpg"),
  },
  {
    id: "10",
    name: "Zwei Kelche",
    image: require("@/assets/images/tarot_cards/Two_Cups.jpg"),
  },
  {
    id: "11",
    name: "Drei Kelche",
    image: require("@/assets/images/tarot_cards/Three_Cups.jpg"),
  },
  {
    id: "12",
    name: "Vier Kelche",
    image: require("@/assets/images/tarot_cards/Four_Cups.jpg"),
  },
  {
    id: "13",
    name: "Fünf Kelche",
    image: require("@/assets/images/tarot_cards/Five_Cups.jpg"),
  },
  {
    id: "14",
    name: "Sechs Kelche",
    image: require("@/assets/images/tarot_cards/Six_Cups.jpg"),
  },
  {
    id: "15",
    name: "Sieben Kelche",
    image: require("@/assets/images/tarot_cards/Seven_Cups.jpg"),
  },
  {
    id: "16",
    name: "Acht Kelche",
    image: require("@/assets/images/tarot_cards/Eight_Cups.jpg"),
  },
  {
    id: "17",
    name: "Neun Kelche",
    image: require("@/assets/images/tarot_cards/Nine_Cups.jpg"),
  },
  {
    id: "18",
    name: "Zehn Kelche",
    image: require("@/assets/images/tarot_cards/Ten_Cups.jpg"),
  },
  {
    id: "19",
    name: "Drei Münzen",
    image: require("@/assets/images/tarot_cards/Three_pents.jpg"),
  },
  {
    id: "20",
    name: "Vier Münzen",
    image: require("@/assets/images/tarot_cards/Four_pents.jpg"),
  },
  {
    id: "21",
    name: "Fünf Münzen",
    image: require("@/assets/images/tarot_cards/Five_pents.jpg"),
  },
  {
    id: "22",
    name: "Sechs Münzen",
    image: require("@/assets/images/tarot_cards/Six_pents.jpg"),
  },
  {
    id: "23",
    name: "Sieben Münzen",
    image: require("@/assets/images/tarot_cards/Seven_pents.jpg"),
  },
  {
    id: "24",
    name: "Acht Münzen",
    image: require("@/assets/images/tarot_cards/Eight_pents.jpg"),
  },
  {
    id: "25",
    name: "Neun Münzen",
    image: require("@/assets/images/tarot_cards/Nine_pents.jpg"),
  },
  {
    id: "26",
    name: "Zehn Münzen",
    image: require("@/assets/images/tarot_cards/Ten_pents.jpg"),
  },
  {
    id: "27",
    name: "Ass Münzen",
    image: require("@/assets/images/tarot_cards/Ace_of_Pentacles.jpg"),
  },
];


