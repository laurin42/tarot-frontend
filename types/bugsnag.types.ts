// Die flexiblere Definition aus bugsnag.ts 端bernehmen
export type BugsnagMetadataValue = string | number | boolean | null | undefined | Record<string, unknown>;
export type BugsnagMetadata = BugsnagMetadataValue | Record<string, BugsnagMetadataValue>;

export interface BugsnagService {
  /** Sendet einen Fehler an Bugsnag */
  notify: (error: Error | string, metadata?: BugsnagMetadata) => unknown;
  /** Erstellt einen Breadcrumb f端r den aktuellen Fehlerkontext */
  leaveBreadcrumb: (message: string, metadata?: BugsnagMetadata) => void;
  /** Setzt Benutzerinformationen f端r Bugsnag-Fehlerberichte */
  setUser: (id?: string, name?: string, email?: string) => void;
  /** Pr端ft, ob Bugsnag erfolgreich initialisiert wurde */
  isStarted: () => boolean;
}