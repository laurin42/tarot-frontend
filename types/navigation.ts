// Entferne die ungenutzten Imports
import type { Href } from "expo-router";

// Wir definieren einen Route-Typ
export type Route = string;

/**
 * Wandelt einen Route-String in einen gültigen Navigationstyp für expo-router um
 * @param route Der zu konvertierende Route-String
 * @returns Ein für expo-router kompatibler Navigationsparameter
 */
export function toValidRoute(route: Route): Href {
  // Dies ist ein Trick, um TypeScript zufriedenzustellen
  // Wir wissen, dass die Routen gültig sind, auch wenn TypeScript es nicht erkennt
  return { pathname: route } as unknown as Href;
}

/**
 * Konstanten für alle App-Routen mit korrekter Typisierung
 */
export const APP_ROUTES = {
  TABS: {
    THREECARDS: "/(tabs)/threecards" as Route,
    DAILYCARD: "/(tabs)/dailycard" as Route,
    PROFILE: "/(tabs)/profile" as Route,
  },
  AUTH: "/(auth)" as Route,
  HOME: "/" as Route,
};