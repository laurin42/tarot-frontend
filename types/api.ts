import { ISelectedAndShownCard } from "@/constants/tarotcards";

export interface ApiErrorResponse {
  error: string;
  status?: number;
}

export interface CardExplanationResponse {
  explanation: string;
  goalsIncluded: boolean;
}

export interface SummaryResponse {
  success: boolean;
  summary: string;
  cards: string[];
  profileInfoIncluded: {
    goals: boolean;
    gender: boolean;
    zodiac: boolean;
    age: boolean;
  };
}

export interface ReadingSummaryResponse {
  id: number;
  name: string;
  description: string;
  sessionId: string;
  position: number;
  createdAt: string;
}