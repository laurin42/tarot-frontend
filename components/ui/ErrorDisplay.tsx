import React from "react";
import { View, Text, Pressable, StyleSheet } from "react-native";
import { devStyles } from "../../styles/devStyles";
import { useRouter } from "expo-router";
import { APP_ROUTES } from "../../constants/routes";
import { toValidRoute } from "../../types/navigation";

interface ErrorDisplayProps {
  message: string;
  showReloadButton?: boolean;
  darkMode?: boolean;
}

export function ErrorDisplay({
  message,
  showReloadButton = false,
  darkMode = false,
}: ErrorDisplayProps): JSX.Element {
  const router = useRouter();

  return (
    <View style={[styles.container, darkMode && styles.darkContainer]}>
      <Text style={[styles.errorText, darkMode && styles.darkModeText]}>
        {message}
      </Text>

      {showReloadButton && (
        <Pressable
          style={devStyles.devButton}
          onPress={() => router.replace(toValidRoute(APP_ROUTES.HOME))}
        >
          <Text style={devStyles.devButtonText}>Neu laden</Text>
        </Pressable>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    padding: 20,
  },
  darkContainer: {
    backgroundColor: "#1F2937",
  },
  errorText: {
    color: "red",
    marginBottom: 20,
    fontSize: 16,
    textAlign: "center",
  },
  darkModeText: {
    color: "white",
  },
});
