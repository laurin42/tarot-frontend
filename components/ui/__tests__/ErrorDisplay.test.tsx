import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import { ErrorDisplay } from "../ErrorDisplay"; // Adjust path if needed
import { APP_ROUTES } from "@/constants/routes"; // Adjust path if needed
import { toValidRoute } from "@/types/navigation"; // Adjust path if needed

// Mock expo-router's useRouter
const mockReplace = jest.fn();
jest.mock("expo-router", () => ({
  useRouter: () => ({
    replace: mockReplace,
  }),
}));

describe("ErrorDisplay Component", () => {
  const errorMessage = "Something went wrong!";

  beforeEach(() => {
    mockReplace.mockClear(); // Clear mock calls before each test
  });

  it("renders the error message correctly", () => {
    render(<ErrorDisplay message={errorMessage} />);
    expect(screen.getByText(errorMessage)).toBeTruthy();
  });

  it("does not show the reload button by default", () => {
    render(<ErrorDisplay message={errorMessage} />);
    expect(screen.queryByText("Neu laden")).toBeNull();
  });

  it("shows the reload button when showReloadButton is true", () => {
    render(<ErrorDisplay message={errorMessage} showReloadButton={true} />);
    expect(screen.getByText("Neu laden")).toBeTruthy();
  });

  it("calls router.replace with the home route when reload button is pressed", () => {
    render(<ErrorDisplay message={errorMessage} showReloadButton={true} />);
    const reloadButton = screen.getByText("Neu laden");
    fireEvent.press(reloadButton);
    expect(mockReplace).toHaveBeenCalledTimes(1);
    expect(mockReplace).toHaveBeenCalledWith(toValidRoute(APP_ROUTES.HOME));
  });

  it("applies dark mode styles when darkMode is true", () => {
    const { getByText, getByTestId } = render(
      // Assuming parent View gets testID or check style directly
      <ErrorDisplay message={errorMessage} darkMode={true} />
    );
    // We can't easily check applied styles directly in RN testing without extra setup.
    // Instead, we rely on snapshot testing or visual regression testing.
    // For unit tests, we'll trust the component logic passes the prop.
    // A snapshot test will capture the different structure/styles if any.
    // Let's add a basic assertion that it renders.
    expect(getByText(errorMessage)).toBeTruthy();
  });

  // Snapshot test
  it("matches snapshot with default props", () => {
    const tree = render(<ErrorDisplay message={errorMessage} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches snapshot with reload button", () => {
    const tree = render(
      <ErrorDisplay message={errorMessage} showReloadButton={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches snapshot with dark mode", () => {
    const tree = render(
      <ErrorDisplay message={errorMessage} darkMode={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches snapshot with dark mode and reload button", () => {
    const tree = render(
      <ErrorDisplay
        message={errorMessage}
        showReloadButton={true}
        darkMode={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
