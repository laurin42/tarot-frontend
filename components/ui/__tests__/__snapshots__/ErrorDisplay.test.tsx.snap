// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay Component matches snapshot with dark mode 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
      {
        "backgroundColor": "#1F2937",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "red",
          "fontSize": 16,
          "marginBottom": 20,
          "textAlign": "center",
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Something went wrong!
  </Text>
</View>
`;

exports[`ErrorDisplay Component matches snapshot with dark mode and reload button 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
      {
        "backgroundColor": "#1F2937",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "red",
          "fontSize": 16,
          "marginBottom": 20,
          "textAlign": "center",
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Something went wrong!
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FF3B30",
        "borderRadius": 20,
        "elevation": 3,
        "paddingHorizontal": 12,
        "paddingVertical": 8,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Neu laden
    </Text>
  </View>
</View>
`;

exports[`ErrorDisplay Component matches snapshot with default props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
      false,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "red",
          "fontSize": 16,
          "marginBottom": 20,
          "textAlign": "center",
        },
        false,
      ]
    }
  >
    Something went wrong!
  </Text>
</View>
`;

exports[`ErrorDisplay Component matches snapshot with reload button 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
      false,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "red",
          "fontSize": 16,
          "marginBottom": 20,
          "textAlign": "center",
        },
        false,
      ]
    }
  >
    Something went wrong!
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FF3B30",
        "borderRadius": 20,
        "elevation": 3,
        "paddingHorizontal": 12,
        "paddingVertical": 8,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Neu laden
    </Text>
  </View>
</View>
`;
